core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/searchfor/ModelSearchImagesUtil.java

```diff
@@ -0,0 +1,138 @@
+/*******************************************************************************
+ * Copyright (c) 2020 THALES GLOBAL SERVICES.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *  
+ * Contributors:
+ *    Thales - initial API and implementation
+ *******************************************************************************/
+package org.polarsys.capella.core.ui.search.searchfor;
+
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.core.runtime.FileLocator;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.jface.resource.ImageDescriptor;
+import org.eclipse.swt.graphics.Image;
+import org.osgi.framework.Bundle;
+
+/**
+ * Utility class for images
+ */
+public final class ModelSearchImagesUtil {

```

Comment:
There are some problems with this class. It not only recreates images but also maintains a map of ImageDescriptor which is already done with the ExtendedImageRegistry. This causes performance issue and code redundancy. Moreover, it's not generic. Capella's elements are treated separately than diagram elements, viewpoint elements, etc.
IMO, there are 2 solutions: 1) Create dummy objects of eClasses just to get their label and image (using EObjectLabelProviderHelper). We've already done this in Capella in org.polarsys.capella.core.ui.metric.actions.MetricAction.
2) More complicated but I think feasible. Contribute a new class (e.g. EClassLabelProviderHelper) to get label and image from EClass. From an EClass we can get the Package URI and then get the ItemProviderAdapterFactory (contributed via org.eclipse.emf.edit.itemProviderAdapterFactories extension point). From the ItemProviderAdapterFactory we can determine the .edit bundle (e.g. org.polarsys.capella.core.data.gen.edit in Capella). With the bundle we can get the image (like what you've done) with ExtendedImageRegistry.INSTANCE.getImageDescriptor. For the label, we can inspire from the method org.eclipse.emf.common.EMFPlugin.InternalHelper.getString(String, boolean).